mongodbatlas/mongodbatlas_test.go

```diff
@@ -153,6 +171,68 @@ func TestNewRequest_withCustomUserAgent(t *testing.T) {
 	}
 }
 
+func TestNewGZipRequest_emptyBody(t *testing.T) {
+	c := NewClient(nil)
+	req, err := c.NewGZipRequest(ctx, http.MethodGet, ".", nil)
+	if err != nil {
+		t.Fatalf("NewRequest returned unexpected error: %v", err)
+	}
+	if req.Body != nil {
+		t.Fatalf("constructed request contains a non-nil Body")
+	}
+}
+
+func TestNewGZipRequest_withUserData(t *testing.T) {
+	c := NewClient(nil)
+
+	requestPath := "foo"
+
+	inURL, outURL := requestPath, defaultBaseURL+requestPath
+	inBody, outBody := &testRequestBody{TestName: "l", TestData: "u"},
```

Comment:
this tests is a lie as you don't expect a json response (nor a json payload as already discussed), let's just make sure that the headers are correct